services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3718,9 +3718,22 @@ Pair<Store, Version> waitVersion(String clusterName, String storeName, int versi
    */
   @Override
   public void setStoreCurrentVersion(String clusterName, String storeName, int versionNumber) {
+    this.setStoreCurrentVersion(clusterName, storeName, versionNumber, false);
+  }
+
+  /**
+   * In most cases, parent region should not update the current version. This is only allowed via an update-store call
+   * where the region filter list only contains one region, which is the region of the parent controller
+   */
+  private void setStoreCurrentVersion(
+      String clusterName,
+      String storeName,
+      int versionNumber,
+      boolean allowedInParent) {
     storeMetadataUpdate(clusterName, storeName, store -> {
-      if (isParent()) {
+      if (isParent() && !allowedInParent) {
         // Parent colo should not update the current version of a store
+        LOGGER.info("Updating current version in parent region is not allowed. Skipping.");
```

Comment:
And add the logging here. Also never add logs without any identifying features, like store/cluster names.